app/PaymentDrivers/Blockonomics/Blockonomics.php

```diff
@@ -132,20 +133,53 @@ public function paymentResponse(PaymentResponseRequest $request)
             'currency' => ['required'],
             'txid' => ['required'],
             'payment_method_id' => ['required'],
+            'btc_address' => ['required'],
+            'btc_amount' => ['required'],
+            'btc_price' => ['required'],
+            // Setting status to required will break the payment process
+            // because sometimes the status is returned as 0 which is falsy
+            // and the validation will fail.
+            // 'status' => ['required'],
         ]);
 
         try {
             $data = [];
-            $fiat_amount = round(($request->btc_price * $request->btc_amount), 2);
+            $fiat_amount = round(($request->btc_price * $request->btc_amount / 100000000), 2);
             $data['amount'] = $fiat_amount;
-            $data['currency'] = $request->currency;
             $data['payment_method_id'] = $request->payment_method_id;
             $data['payment_type'] = PaymentType::CRYPTO;
             $data['gateway_type_id'] = GatewayType::CRYPTO;
-            $data['transaction_reference'] = $request->txid;
 
-            $statusId = Payment::STATUS_PENDING;
+
+            define('TEST_TXID', 'WarningThisIsAGeneratedTestPaymentAndNotARealBitcoinTransaction');
```

Comment:
Just as a standard, can we use a private class property rather than a define() like this please.

Reply:
Good call, now it's been updated!